### PR TITLE
Migrate `src/{main,common,config}/**` to ES Modules

### PR DESCRIPTION
**Pull request by fisker**
## Description

Part of #10157

## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/common/create-ignorer.js [hunk 1]
```diff
@@ -1,8 +1,6 @@
-"use strict";
-
-const path = require("path");
-const ignore = require("ignore");
-const getFileContentOrNull = require("../utils/get-file-content-or-null.js");
+import path from "path";
```
Comment (sosukesuzuki) [inline src/common/create-ignorer.js:1]: Can we use `node` protocol in `src`?

**Review by sosukesuzuki — COMMENTED** [review]
Thanks for doing great work. One question

File: src/common/create-ignorer.js (hunk 1 above)
Comment (fisker) [reply src/common/create-ignorer.js:1]: https://github.com/prettier/prettier/pull/11903/files#diff-e2954b558f2aa82baff0e30964490d12942e0e251c1aa56c3294de6ec67b7cf5R182, didn't dig yet.  Let's improve it later, simply run eslint will fix that.

Comment (fisker) [reply src/common/create-ignorer.js:1]: There are lot's of TODOs, I'll try list them after we merge all those PRs.

Comment (sosukesuzuki) [reply src/common/create-ignorer.js:1]: Okay. Thanks

**Review by sosukesuzuki — APPROVED** [review]

**Pull request merged**